### ISSUE TITLE
AAAD fix for Oppo/OnePLus

### ISSUE THREAD

**Issue by Pepilton** [shmykelsa/AAAD#235]
Google package installer ist not working - at least on my Oppo Find X3Pro. I forced stop of the package installer and installed googles package installer from your link. First installation over google package installer nothing happens. With the second try message pops up that package installer was ended ... followed exactely the instructions. Any idea?

**Comment by shmykelsa** [comment]
Please include:

- Android version
- AAAD version
- Exact steps you take to reproduce the issue

**Comment by BatlleSonic** [comment]
I have the Same issue by the Oppo Find X5 pro

Android version : 13 (Build CPH2305VF_11_C.42)
AAAD version: 1.4.4

Exact steps you take to reproduce the issue:
When I download the app via AAAD and then select the non-standard installer, nothing visible happens. Unfortunately, also no entry for the installed app. If I repeat the process with the "standard" installer, I get the message Cancel or Update. Unfortunately, I can not uninstall the app, because it is not listed in the App Manager. It is also not listed in Android Auto.

**Comment by shmykelsa** [comment]
> When I download the app (AA Mirror) via AAAD and then select the non-default installer, nothing visible happens. 

Can you confirm that you are **not** pressing the green one to begin installation?

> Unfortunately, no entry for the installed app (AA Mirror) either. When I repeat the process with the "default" installer, I get a Cancel or Update message. 

This makes me think you already have the app installed with a different signature, which provokes a signature mismatch and installation error.


> Unfortunately, I cannot uninstall the app (AA Mirror) because it is not listed in the app manager. It is also not listed in Android Auto.

This is not possible at all. Each and every package must appear in apps' manager in system's settings.

**Comment by BatlleSonic** [comment]
> Can you confirm that you are not pressing the green button to start the installation?

Yes, and since then there is the problem where you suspect that something is wrong with the signature. 

> This leads me to believe that you have already installed the application with a different signature, resulting in a signature mismatch and an installation error.


No, the problem exists only since the installation with the installer from them. 


> This is not possible at all. Every single package has to be displayed in the App Manager in the system settings.

Unfortunately, nothing is entered with their installer but something seems to be installed. Just incomplete and without regestrierung.

> If that is not the case, there is definitely something wrong with your phone, which probably requires a factory reset.


I will do this on occasion and will report back. But as I said, that problem came with the installer from them.

**Comment by Pepilton** [comment]
> Please include:
> 
> * Android version
13 (Color OS 13.0 Version CPH2173_11_F.80)
> * AAAD version
1.4.4 (Android Auto 9.0.130744-release.daily)
> * Exact steps you take to reproduce the issue
Force stop for standard package installer via app management (@BatlleSonic you probably have to activate "show system-apps)
Downloaded and installes google package installer from your link
Downloaded Carstream via AAAD
When donwloaded finished to potions are available, standard package installer an a new one
Choosing the new one nothing visible happens. Trying again, i get a cancel message
With standard installer everthing works fine. Carsstream ist visible in the AA Launcher, but getting the "no messages during drive"-notice.

**Comment by BatlleSonic** [comment]
@Pepilton  
yes even under "SystemApps" it does not show me the (for example) Mirror AA app. 

The installation of the "modified" installer I have been able to perform as described in the instructions

**Comment by Pepilton** [comment]
> @Pepilton yes even under "SystemApps" it does not show me the (for example) Mirror AA app.
> 
> The installation of the "modified" installer I have been able to perform as described in the instructions

Same for me and the same problem with FindX 3 Pro, despite your issues with the standard installer. This wprks fine for me but gettin the no messages during drive"-notice. Guess we have the same problem and you an additional one. ;-)

**Comment by Pepilton** [comment]
@shmykelsa is there any solution that the Google package installer is not working on Oppo X3/X5?

**Comment by shmykelsa** [comment]
> Yes, and since then there is the problem where you suspect that something is wrong with the signature.

Again, you must **not** press the green installation manager. It's either a brown or a grey one.


> @Pepilton yes even under "SystemApps" it does not show me the (for example) Mirror AA app.

That would be correct. AA Mirror is not a system app. If it's not shown at all, and the app is installed, then you have something wrong on your phone, and I'd suggest a factory reset.


> @shmykelsa is there any solution that google package installer 3.0.269 is not working on Oppo X3/X5?

It has been tested on Oppo X3 and it works. You might want to double check the instructions and try again. Or the issue is something bigger than AAAD.

**Comment by naobug** [comment]
I have the same problem. Use default package installer installs the car stream app or fermata auto but shows "no messages during this drive".
Using the fixed installer does nothing, no reaction, no app installed. 

Realme X2 Pro 
Android 11
AAAD 1.4.4

**Comment by shmykelsa** [comment]
@naobug 

Were the apps you tried already installed? Have you tried uninstalling before installing with new method? And to be absolutely clear: have you made sure you **did not** press the installer that has the green Android robot logo as per instructions given?

**Comment by naobug** [comment]
1) I made sure that the app was not installed before.
2) I clicked the green android robot because that is the new icon that appeared after the fix was installed. The default installer that I needed to force stop has the green icon with the white downward pointing arrow.

**Comment by naobug** [comment]
![Screenshot_2023-03-12-13-39-02-49_fc704e6b13c4fb26bf5e411f75da84f2](https://user-images.githubusercontent.com/12033198/224545237-ab622fed-fe7a-4ae5-a388-262d549be92d.jpg)
![Screenshot_2023-03-12-13-39-18-40_c31b32364ce19ca8fcd150a417ecce58](https://user-images.githubusercontent.com/12033198/224545242-bd76923f-1f75-4348-9aa0-a5fb406d22e4.jpg)

**Comment by naobug** [comment]
The only other thing that I found is that people back in 2021 (maybe those with Realme devices) got rid of the problem by installing the old Android Auto Version 6.3. (so for them it started with AA 6.4). But this isn't a viable solution.

I am using Android Auto 8.9.630624 together with AAAD 1.4.4 on Android 11.

**Comment by gabrys** [comment]
I have a variation of this error. I cannot install the modified google installer, I'm getting the following error: App not installed as package conflicts with an existing package.

Os is LineageOS nightly, OnePlus 9 pro, Android 13.

![Screenshot_20230312-215459_Package installer](https://user-images.githubusercontent.com/74567/224573570-5f9efa8c-0ca0-4734-b8c1-0497eb53592e.png)

**Comment by shmykelsa** [comment]
> I have a variation of this error. I cannot install the modified google installer, I'm getting the following error: App not installed as package conflicts with an existing package.
> 
> 
> 
> Os is LineageOS nightly, OnePlus 9 pro, Android 13.
> 

I suppose you are rooted?

**Comment by gabrys** [comment]
@shmykelsa I'm not rooted as in SuperSU installed, bit I can access the root she'll via adb (haven't done that though)

**Comment by shmykelsa** [comment]
> @shmykelsa I'm not rooted as in SuperSU installed, bit I can access the root she'll via adb (haven't done that though)

A more reliable solution for rooted users would be patching the apps via [AA AIO TWEAKER](https://github.com/shmykelsa/AA-Tweaker). I'd suggest looking into it.

**Comment by gabrys** [comment]
When I installed the tweaker app it didn't request root. I believe I don't have a rooted OS (beyond adb shell) and I don't know how to make it so right now.

It was relatively easy with CyanogenMod and LineageOS in the past (either built-in or a simple galah away), but right now I don't see any official/documented way of installing SuperSU or equivalent. I'm not even sure if SuperSU is even the preferred root app these days. 

I was really happy when I discovered AAAD and AA Mirror, thank you for your work on this! I wish this worked with my new phone as well 😂

**Comment by shmykelsa** [comment]
> When I installed the tweaker app it didn't request root. I believe I don't have a rooted OS (beyond adb shell) and I don't know how to make it so right now.
> 
> 
> 
> It was relatively easy with CyanogenMod and LineageOS in the past (either built-in or a simple galah away), but right now I don't see any official/documented way of installing SuperSU or equivalent. I'm not even sure if SuperSU is even the preferred root app these days. 
> 

I suggest looking into Magisk which is now the golden standard for rooting. You might want to start from extracting boot.img from recovery then patch it through Magisk manager and then write it to the boot.img partition. If I recall correctly you can also rename the .apk to .zip and flash it through recovery right away.

> 
> 
> I was really happy when I discovered AAAD and AA Mirror, thank you for your work on this! I wish this worked with my new phone as well 😂

With root you might even experience a better mirroring on Android Auto as permissions can be set permanently. Screen2Auto is a great example of an app that works much better with root permissions.

**Comment by Pepilton** [comment]
> > Yes, and since then there is the problem where you suspect that something is wrong with the signature.
> 
> Again, you must **not** press the green installation manager. It's either a brown or a grey one.
> 
> > @Pepilton yes even under "SystemApps" it does not show me the (for example) Mirror AA app.
> 
> That would be correct. AA Mirror is not a system app. If it's not shown at all, and the app is installed, then you have something wrong on your phone, and I'd suggest a factory reset.
> 
> > @shmykelsa is there any solution that google package installer 3.0.269 is not working on Oppo X3/X5?
> 
> It has been tested on Oppo X3 and it works. You might want to double check the instructions and try again. Or the issue is something bigger than AAAD.

Using Oppo X3 and double checked your instructions. Uninstalled Carstream and Google Package Installer, forced Package Installer to stop, installed new one, installed Carstream but still getting the "No new Messages". Only thing is my carstream APK was downloaded a few weeks ago from AAAD. So the apk already exists on my phone. Could that be the issue? Have no downloads left for this month. APK is version 2.0.5.

**Comment by naobug** [comment]
I guess the icon depends on the phone; here is my Realme

![Screenshot_2023-03-12-13-39-18-40_c31b32364ce19ca8fcd150a417ecce58](https://user-images.githubusercontent.com/12033198/224841176-9358b467-b0eb-42c8-8d88-7fcdbe11355b.jpg)

**Comment by shmykelsa** [comment]
> Using Oppo X3 and double checked your instructions. Uninstalled Carstream and Google Package Installer, forced Package Installer to stop, installed packagef installer from your link again, installed Carstream but still getting the "No new Messages". 

Just installing the package installer does not fix the issue if the apps are already installed.

> Only thing is my carstream APK was downloaded a few weeks ago from AAAD. So the apk already exists on my phone.

Definitely yes

**Comment by shmykelsa** [comment]
@naobug contact help team through the app by going in the top right menu and hit "Help". We'll make sure your download is reset so you can try again.

**Comment by Pepilton** [comment]
> @naobug contact help team through the app by going in the top right menu and hit "Help". We'll make sure your download is reset so you can try again.

@shmykelsa is that possible for me too, because i also wasted my download with using the wrong package installer ...

**Comment by shmykelsa** [comment]
> > @naobug contact help team through the app by going in the top right menu and hit "Help". We'll make sure your download is reset so you can try again.
> 
> @shmykelsa is that possible for me too, because i also wasted my download with using the wrong package installer ...

Sure thing

**Comment by EazyDefcon** [comment]
As gabrys, I also get "App not installed as package conflicts with an existing package." When trying to install Package Installer. Have forced closed but I wont help. Perhaps freezing or uninstall the package installer app with adb?

Btw. My installed Package installer apps icon is brown.

Have tried a thousand times with different howtos to get screen 2 auto but nothing works. 
Last thing to try is root my phone, but I wont do that. Guess I have to give up if rooting is necessary  :(

Oneplus 7 t Pro (No root)
Rom: crDroid 9 (Android 13)
AA version: 8.8.630424

**Comment by mhmtnasir** [comment]
Oppo and OnePlus have this issue, try KingInstaller with the APK files themselves

**Comment by pdatta1973** [comment]
I have the Oppo Reno 2Z and trying to install AAAD and Carstream apps on my mobile and failed everytime I did till yesterday late night. I tried all the said and suggested steps for installing the above softwares but none working so far. I too tried with XAPK and KingInstaller apps too bu the same fate. All the apps starts installing and vanishes without any said error and no installations. 

Phone Specifications- OPPO Reno2z (CPH1945_11_F.65)
RAM - 13GB / 256 GB 
OS- ColorOS Version - V11.1

Please help
